developer/build_plugin.py

```diff
@@ -143,32 +143,66 @@ def _process_configuration(sg_connection, source_path, target_path, bundle_cache
                 "version": BAKED_BUNDLE_VERSION
             }
         )
-
+        # Workaround for tk-core bootstrap needing a shotgun.yml file: create
+        # one with an "unspecified" host.
+        shotgun_yaml_path = os.path.join(install_path, "config", "core", "shotgun.yml")
+        if not os.path.exists(shotgun_yaml_path):
+            logger.info("Patching %s" % shotgun_yaml_path)
+            with open(shotgun_yaml_path, "w") as pf:
+                pf.write("#Workaround for tk-core bootstrap\nhost: unspecified")
     else:
 
         # if the descriptor in the config contains a version number
         # we will go into a fixed update mode.
-        if is_descriptor_version_missing(base_config_uri_dict):
-            logger.info(
-                "Your configuration definition does not contain a version number. "
-                "This means that the plugin will attempt to auto update at startup."
-            )
-            using_latest_config = True
+        using_latest_config = is_descriptor_version_missing(base_config_uri_dict)
+        if not do_bake:
+            if using_latest_config:
+                logger.info(
+                    "Your configuration definition does not contain a version number. "
+                    "This means that the plugin will attempt to auto update at startup."
+                )
+            else:
+                logger.info(
+                    "Your configuration definition contains a version number. "
+                    "This means that the plugin will be frozen and no automatic updates "
+                    "will be performed at startup."
+                )
         else:
             logger.info(
-                "Your configuration definition contains a version number. "
+                "Baking your configuration definition into an immutable state. "
                 "This means that the plugin will be frozen and no automatic updates "
                 "will be performed at startup."
             )
-            using_latest_config = False
+            if using_latest_config:
+                logger.info(
+                    "Your configuration definition does not contain a version number. "
+                    "Retrieving the latest version of the configuration for baking."
+                )
 
         cfg_descriptor = create_descriptor(
             sg_connection,
             Descriptor.CONFIG,
             base_config_uri_dict,
             resolve_latest=using_latest_config
         )
-
+        if do_bake:
+            cfg_descriptor.ensure_local()
+            local_path = cfg_descriptor.get_path()
+            if not local_path:
+                raise ValueError("Unable to get a local copy of %s" % cfg_descriptor)
+            baked_descriptor = {
+                "type": bootstrap_constants.BAKED_DESCRIPTOR_TYPE,
+                "path": local_path
+            }
+            manifest_data["base_configuration"] = baked_descriptor
```

Comment:
I think we should also make sure that `info.yml` is not copied across when in bake mode.